examples/dam.jl

```diff
@@ -0,0 +1,163 @@
+#  Copyright 2015, Vincent Leclere, Francois Pacaud and Henri Gerard
+#  This Source Code Form is subject to the terms of the Mozilla Public
+#  License, v. 2.0. If a copy of the MPL was not distributed with this
+#  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#############################################################################
+# Test SDDP with dam example
+# Source: Adrien Cassegrain
+#############################################################################
+
+
+include("../src/SDDP.jl")
+include("../src/SDDPoptimize.jl")
+include("../src/simulate.jl")
+
+using JuMP
+using CPLEX
+
+N_STAGES = 52
+N_SCENARIOS = 1
+
+alea_year = Array([7.0 7.0 8.0 3.0 1.0 1.0 3.0 4.0 3.0 2.0 6.0 5.0 2.0 6.0 4.0 7.0 3.0 4.0 1.0 1.0 6.0 2.0 2.0 8.0 3.0 7.0 3.0 1.0 4.0 2.0 4.0 1.0 3.0 2.0 8.0 1.0 5.0 5.0 2.0 1.0 6.0 7.0 5.0 1.0 7.0 7.0 7.0 4.0 3.0 2.0 8.0 7.0])
+
+
+# FINAL TIME:
+TF = 52
+
+# COST:
+COST = -66*2.7*(1 + .5*(rand(TF) - .5))
+
+# Constants:
+VOLUME_MAX = 100
+VOLUME_MIN = 0
+
+CONTROL_MAX = round(Int, .4/7. * VOLUME_MAX) + 1
+CONTROL_MIN = 0
+
+W_MAX = round(Int, .5/7. * VOLUME_MAX)
+W_MIN = 0
+DW = 1
+
+T0 = 1
+HORIZON = 52
+
+
+# Define aleas' space:
+N_ALEAS = Int(round(Int, (W_MAX - W_MIN) / DW + 1))
+ALEAS = linspace(W_MIN, W_MAX, N_ALEAS)
+
+
+# Define dynamic of the dam:
+function dynamic(x, u, w)
+    return x[1] - u[1] - u[2] + w[1]
+end
+
+# Define cost corresponding to each timestep:
+function cost_t(t, x, u, w)
+    return COST[t] * u[1]
+end
+
+
+"""Solve the problem with a solver, supposing the aleas are known
+in advance."""
+function solve_determinist_problem()
+    println(alea_year)
+    m = Model(solver=CplexSolver(CPX_PARAM_SIMDISPLAY=0))
```

Comment:
Put Clp here to be coherent. I think it should be great to define a function which set Clpsolver, or Cplex solver instead of changing the code when we want to change of solver